Automattic/wp-calypso
# Suspended Domains i1: Show warning before launching site with custom domain and unverified email

Closes https://github.com/Automattic/dotcom-forge/issues/3949

## Proposed Changes

* Title

![image](https://github.com/Automattic/wp-calypso/assets/6851384/a46d7615-27d5-4c95-96d8-ea7eb7a372e1)


## Testing Instructions



You can test it with a new account and domain purchases, but I found  changing `isEmailVerified` to be easiest. 

* Enable the test store and sandbox the API. Use a test credit card for checkout
* Go through one of the start/setup flows that client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.tsx, such as Link in Bio, Newsletter, or blog, and buy a custom domain choose "Skip to dashboard". 
* During checkout, use an unverified email address in the domain contact info
* Progress to Launchpad and then try to complete setup. 
* Confirm a dialog is shown as above. Hit cancel. 
* Try again and confirm the dialog is reshown. Hit "Continue anyway". 

* Try to checkout with a verified email address and confirm the dialog is never shown. 

## Pre-merge Checklist
